.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,245 @@
+name: Build and Upload Artifacts
+
+on:
+  workflow_dispatch:
+    inputs:
+      partner_chains_smart_contracts_sha:
+        description: "partner-chains-smart-contracts commit SHA or branch to build from"
+      partner-chains-smart-contracts-tag:
+        description: "partner-chains-smart-contract release tag to append to artifact name (optional)"
+        required: false
+      partner_chains_sha:
+        description: "partner-chains commit SHA or branch to build from"
+      partner-chains-tag:
+        description: "partner-chains release tag to append to artifact name (optional)"
+        required: false
+
+jobs:
+  partner-chains-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set filename variables
+        id: set-filenames
+        run: |
+          if [ -n "${{ github.event.inputs.partner-chains-tag }}" ]; then
+            echo "PARTNER_CHAINS_CLI_X86_64_LINUX=partner-chains-cli-${{ github.event.inputs.partner-chains-tag }}-x86_64-linux" >> $GITHUB_ENV
+            echo "PARTNER_CHAINS_NODE_X86_64_LINUX=partner-chains-node-${{ github.event.inputs.partner-chains-tag }}-x86_64-linux" >> $GITHUB_ENV
+          else
+            echo "PARTNER_CHAINS_CLI_X86_64_LINUX=partner-chains-cli-x86_64-linux" >> $GITHUB_ENV
+            echo "PARTNER_CHAINS_NODE_X86_64_LINUX=partner-chains-node-x86_64-linux" >> $GITHUB_ENV
+          fi
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.partner_chains_sha }}
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```

Comment:
I'm afraid it could backfire at some point, because we have toolchain defined implicitly in Earthfile and explicitly in rust-toolchain.yaml. There is a potential for a hard to understand bug when locally and earthly-ci would builds the code but this action fails due to stricter compiler behavior.

It is not an unlikely scenario. Just yesterday Michael reported that polkadot-sdk fails to build with rust 1.80 but is okay with 1.77.

Reply:
Ah my bad! (not a Rust developer)

It seems cargo builds automatically use the `rust-toolchain.toml` if it exists, so I have removed those steps entirely